docs/ref/contrib/gis/gdal.txt

```diff
@@ -1096,35 +1099,101 @@ one or more layers of data named bands.  Each band, represented by a
 image is represented as three bands: one for red, one for green, and one for
 blue.
 
-.. class:: GDALRaster(ds_input)
+.. note::
 
-    The constructor for ``GDALRaster`` accepts a single parameter: the path of
-    the file you want to read.
+     For raster data there is no difference between a raster instance and its
+     data source. Unlike for the Geometries objects, :class:`GDALRaster` objects are
+     always a data source. Temporary rasters can be instantiated in memory
+     using the corresponding driver, but will be of the same class as file-based
+     raster sources.
+
+.. class:: GDALRaster(ds_input, write=False)
+
+    The constructor for ``GDALRaster`` accepts two parameters. The first parameter
+    defines the raster source, it is either a path to a file or spatial data with
+    values defining the properties of a new raster (such as size and name). If the
+    input is a file path, the second parameter specifies if the raster should
+    be opened with write access. The following example shows how rasters can be
+    created from different input sources (using the sample data from the GeoDjango
+    tests, see the :ref:`sample_data` section)::
+
+      >>> from django.contrib.gis.gdal.raster.source import GDALRaster
+      >>> rst = GDALRaster('/path/to/your/raster.tif', write=False)
+      >>> rst.name
+      '/path/to/your/raster.tif'
```

Comment:
Adopted this to new sample data retrieval. Also added crossref above to new link for the raster file download in Sample Data section.